src/manager/agent.manager.service.ts

```diff
@@ -92,19 +92,17 @@ export class AgentManagerService {
             connectionData = await this.createConnection(agentId, adminPort, adminApiKey);
         }
 
-        return { agentId, containerId, adminPort, httpPort, connectionData };
+        return { agentId, connectionData };
     }
 
     /**
      * TODO we should probably respond with something
      */
     public async spinDownAgent(agentId: string) {
-        // TODO: do we need to remove this entry
-        const agent: any = await this.cache.get(agentId);
+        Logger.log('Spinning down agent', agentId);
         await this.cache.del(agentId);
```

Comment:
Same with this .del

Reply:
We need to cache adminAPIKey for the governance Policy handlers.  Therefore we have to delete it on stopAgent